### PR TITLE
fixing issue with executable items count

### PR DESCRIPTION
**Pull request by sudarshan0212**
Thank you for your contribution.
Before submitting this PR, please make sure:
- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is point to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for Existing Functionality
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments

### CODE REVIEW

File: Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs [hunk 1]
```diff
@@ -294,7 +294,7 @@ public static AccountReportRunner MapRunnerStartData(GingerRunner gingerRunner,
             accountReportRunner.EnvironmentId = gingerRunner.ProjEnvironment.Guid;
             accountReportRunner.StartTimeStamp = gingerRunner.Executor.StartTimeStamp;
             accountReportRunner.ApplicationAgentsMappingList = gingerRunner.ApplicationAgents.Select(a => a.AgentName + "_:_" + a.AppName).ToList();
-            SetRunnerChildCounts((GingerExecutionEngine)gingerRunner.Executor, accountReportRunner);
+            SetRunnerChildCounts((GingerExecutionEngine)gingerRunner.Executor, accountReportRunner, true);
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs:297]: Changes needs to be added to the "Official Release" branch and not to master for being part of 5.1 release

File: Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs [hunk 2]
```diff
@@ -424,15 +424,29 @@ private static void SetRunnerChildCounts(GingerExecutionEngine runner, AccountRe
 
             foreach (BusinessFlow businessFlow in runner.BusinessFlows)
             {
-                ChildExecutableItemsCountActivity = ChildExecutableItemsCountActivity + businessFlow.Activities.Count(x => x.Active && (x.Status == eRunStatus.Passed || x.Status == eRunStatus.FailIgnored || x.Status == eRunStatus.Failed || x.Status == eRunStatus.Blocked));
+                if(IsStart)
+                {
+                    ChildExecutableItemsCountActivity = ChildExecutableItemsCountActivity + businessFlow.Activities.Count(x=> x.Active);
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs:429]: are we now collecting "Executed" count on Start for calculating the progress during execution?
how we did it before?

File: Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs [hunk 3]
```diff
@@ -424,15 +424,29 @@ private static void SetRunnerChildCounts(GingerExecutionEngine runner, AccountRe
...
+                }
+                else
+                {
+                    ChildExecutableItemsCountActivity = ChildExecutableItemsCountActivity + businessFlow.Activities.Count(x => x.Active && (x.Status == eRunStatus.Passed || x.Status == eRunStatus.FailIgnored || x.Status == eRunStatus.Failed || x.Status == eRunStatus.Blocked));
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs:433]: what is different between the calculation of ChildExecutableItemsCountActivity in line 433 and in line 436?

File: Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs [hunk 4]
```diff
@@ -424,15 +424,29 @@ private static void SetRunnerChildCounts(GingerExecutionEngine runner, AccountRe
...
                 ChildExecutedItemsCountActivity = ChildExecutedItemsCountActivity + businessFlow.Activities.Count(ac => ac.Status == eRunStatus.Failed || ac.Status == eRunStatus.Passed || ac.Status == eRunStatus.FailIgnored || ac.Status == eRunStatus.Stopped || ac.Status == eRunStatus.Completed);
 
                 ChildPassedItemsCountActivity = ChildPassedItemsCountActivity + businessFlow.Activities.Count(ac => ac.Status == eRunStatus.Passed);
 
                 foreach (Activity activity in businessFlow.Activities)
                 {
-                    ChildExecutableItemsCountAction = ChildExecutableItemsCountAction + activity.Acts.Count(x => x.Active && (x.Status == eRunStatus.Passed || x.Status == eRunStatus.FailIgnored || x.Status == eRunStatus.Failed || x.Status == eRunStatus.Blocked));
+                    if(IsStart)
+                    {
+                        ChildExecutableItemsCountAction = ChildExecutableItemsCountAction + activity.Acts.Count;
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs:444]: no need to consider only the Active Actions?

**Review by MeniKadosh1 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs (hunk 2 above)
Comment (sudarshan0212) [reply Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs:429]: for calculation of count at start & end we were using same function which is bug.

File: Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs (hunk 3 above)
Comment (sudarshan0212) [reply Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs:433]: line no 433 (ChildExecutableItemsCountActivity) is total activity count and line no 436(ChildExecutedItemsCountActivity) is total executed activity count.

File: Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs (hunk 4 above)
Comment (sudarshan0212) [reply Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs]: Updated code.

**Review by MeniKadosh1 — APPROVED** [review]

**Pull request merged**